src/operator/tensor/cast_storage-inl.h

```diff
@@ -120,9 +119,13 @@ struct CastStorageRspDnsKernel {
     IType rid = idx[i];
     dim_t dns_offset = rid * row_length;
     dim_t rsp_offset = i * row_length;
-    for (dim_t col = 0; col < row_length; col++) {
-      dns[dns_offset + col] = data[rsp_offset + col];
-    }
+    #if defined(__CUDACC__)
+      for (dim_t col = 0; col < row_length; col++) {
+        dns[dns_offset + col] = data[rsp_offset + col];
+      }
+    #else
+      memcpy(dns + dns_offset, data + rsp_offset, sizeof(DType) * row_length);
```

Comment:
@piiswrong do we want to allow/expose the direct use of `memcpy` at operator level? Do we want to use some mxnet wrapper around it? 
There's also `__builtin_memcpy ` which calls normal `memcpy` for large buffers and a faster version for small buffers. Shall we have sth like this to enable the use of `__builtin_memcpy` when turned on? 
```
#ifdef MX_USE_BUILTIN_MEMCPY
#define MX_MEMCPY __builtin_memcpy
#else 
#define MX_MEMCPY memcpy 
```

Reply:
Yes we need to have a mxnet wrapper.

Reply:
I could find some other files direct use `memcpy`. Should I change them in this pr?

Reply:
@piiswrong any concern for changing all `memcpy` occurrences?

Reply:
Hi, I found that the ci-build always failed because the connection failed.
Ex.
`Remote call on JNLP4-connect connection from ec2-52-42-174-139.us-west-2.compute.amazonaws.com/52.42.174.139:53189 failed`

Reply:
Sorry about that. The CI is not very stable recently.